### PR TITLE
Assign VN for COMMA in gtWrapWithSideEffects

### PR DESCRIPTION
**Pull request by EgorBo**
Let's see if this works out on CI..

### CODE REVIEW

**Comment by EgorBo** [comment]
@MihuBot -dependsOn 108838

**Comment by EgorBo** [comment]
PTAL @AndyAyersMS @dotnet/jit-contrib 
I expected more [diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=845917&view=ms.vss-build-web.run-extensions-tab), but there are just a few. The main motivation was to optimize bound checks for FOH-allocated arrays, repro:
```cs
using System;
using System.Runtime.CompilerServices;

class Program
{
    static void Main()
    {
        for (int i=0; i<200; i++) {
            Test();
            Thread.Sleep(16);
        }
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    static int Test() => MyArray.arr[2]; // should drop bound checks, array length is known
}

class MyArray {
    public static readonly int[] arr = new int[10];
}
```

at least, it's a clean up..

File: src/coreclr/jit/gentree.cpp [hunk 1]
```diff
@@ -17223,7 +17223,9 @@ GenTree* Compiler::gtWrapWithSideEffects(GenTree*     tree,
         // It should be possible to be smarter here and allow such cases by extracting the side effects
         // properly for this particular case. For now, caller is responsible for avoiding such cases.
 
-        tree = gtNewOperNode(GT_COMMA, tree->TypeGet(), sideEffects, tree);
+        GenTree* comma  = gtNewOperNode(GT_COMMA, tree->TypeGet(), sideEffects, tree);
+        comma->gtVNPair = tree->gtVNPair;
+        return comma;
```
Comment (SingleAccretion) [inline src/coreclr/jit/gentree.cpp:17228]: This needs to attach the exception set from `sideEffects`.

Comment (EgorBo) [reply src/coreclr/jit/gentree.cpp]: Thanks! Addressed

**Review by AndyAyersMS — APPROVED** [review]

**Comment by MichalPetryka** [comment]
@MihuBot -dependsOn 108838,108606

**Comment by MichalPetryka** [comment]
> I expected more [diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=845917&view=ms.vss-build-web.run-extensions-tab), but there are just a few.

See above for more diffs with #108606

**Comment by EgorBo** [comment]
SPMI failures are due to JIT-EE bump in a PR merged recently

**Pull request merged**